File: template/frontend/src/mocks/store.ts
import type { CreateTaskBody, Task, UpdateTaskBody } from "@/api/types";

const SEED: readonly Task[] = [
  { id: "1", title: "Read AGENTS.md", done: true },
  { id: "2", title: "Run the app in mock mode", done: false },
  { id: "3", title: "Replace this demo with a real feature", done: false },
];

export interface TaskStore {
  list: () => Task[];
  create: (body: CreateTaskBody) => Task;
  update: (id: string, body: UpdateTaskBody) => Task | null;
  remove: (id: string) => boolean;
  reset: () => void;
}

export function createTaskStore(seed: readonly Task[] = SEED): TaskStore {
  let tasks: Task[] = [...seed];
  let nextId = seed.length + 1;

  return {
    list: () => [...tasks],
    create: ({ title }) => {
      const task: Task = { id: String(nextId), title, done: false };
      nextId += 1;
      tasks = [...tasks, task];
      return task;
    },
    update: (id, { done }) => {
      const existing = tasks.find((task) => task.id === id);
      if (existing === undefined) {
        return null;
      }
      const updated: Task = { ...existing, done };
      tasks = tasks.map((task) => (task.id === id ? updated : task));
      return updated;
    },
    remove: (id) => {
      const before = tasks.length;
      tasks = tasks.filter((task) => task.id !== id);
      return tasks.length < before;
    },
    reset: () => {
      tasks = [...seed];
      nextId = seed.length + 1;
    },
  };
}

export const taskStore = createTaskStore();

File: template/backend/app/store.py
from app.models import CreateTaskBody, Task, UpdateTaskBody

SEED = (
    ("1", "Read AGENTS.md", True),
    ("2", "Run the app in mock mode", False),
    ("3", "Replace this demo with a real feature", False),
)


class TaskStore:
    _tasks: list[Task]
    _next_id: int

    def __init__(self) -> None:
        self.reset()

    def reset(self) -> None:
        self._tasks = [Task(id=id, title=title, done=done) for id, title, done in SEED]
        self._next_id = len(SEED) + 1

    def list(self) -> list[Task]:
        return list(self._tasks)

    def create(self, body: CreateTaskBody) -> Task:
        task = Task(id=str(self._next_id), title=body.title, done=False)
        self._next_id += 1
        self._tasks.append(task)
        return task

    def update(self, id: str, body: UpdateTaskBody) -> Task | None:
        for task in self._tasks:
            if task.id == id:
                task.done = body.done
                return task
        return None

    def remove(self, id: str) -> bool:
        for index, task in enumerate(self._tasks):
            if task.id == id:
                del self._tasks[index]
                return True
        return False


task_store = TaskStore()
